test/integration/style-spec/validate_spec.test.ts

```diff
@@ -19,10 +19,80 @@ describe('validate_spec', () => {
     });
 
     test('errors from validate do not contain line numbers', () => {
-        const style = JSON.parse(fs.readFileSync('test/integration/style-spec/tests/bad-color.input.json', 'utf8'));
+        const style = JSON.parse(
+            fs.readFileSync(
+                'test/integration/style-spec/tests/bad-color.input.json',
+                'utf8'
+            )
+        );
 
         const result = validate(style, reference);
         expect(result[0].line).toBeUndefined();
     });
 
+    test('Validate sdk-support in spec', () => {
+        const issueTrackers = {
+            js: 'https://github.com/maplibre/maplibre-gl-js/issues',
+            android: 'https://github.com/maplibre/maplibre-native/issues',
+            ios: 'https://github.com/maplibre/maplibre-native/issues',
+        };
+        const platforms = Object.keys(issueTrackers);
+
+        function validatePlatforms(platformSupport, path) {
+            const notSupportedOnAnyPlatform = platforms.every(
+                (platform) => !platformSupport[platform]
+            );
+
+            for (const platform of platforms) {
+                if (notSupportedOnAnyPlatform) {
+                    // don't consider it a problem is something is not supported on any platform
+                    return;
+                }
+                if (!platformSupport[platform])
+                    throw new Error(`Missing platform '${platform}' in sdk-support at ${path.join(
+                        '.'
+                    )}.
+Please create a tracking issue and add the link.`);
+
+                const maplibreIssue =
+          /https:\/\/github.com\/maplibre\/[^/]+\/issues\/(\d+)/;
+                const version = /^\d+\.\d+\.\d+$/;
+                const values = new Set(['supported', 'wontfix']);
+
+                const support = platformSupport[platform];
+                if (
+                    !support.match(maplibreIssue) &&
+          !support.match(version) &&
+          !values.has(support)
+                ) {
+                    throw new Error(
+                        `Unsuppored sdk-support value '${support}'. Use one of the following\n` +
+              '- If supported: version number (e.g. 1.0.0) to indicate support since this version (or "supported" to indicate it has always been supported)\n' +
+              '- If it will never be supported: "wontfix" to indicate it will never be supported\n' +
+              '- A link to a tracking issue'
+                    );
+                }
+            }
+        }
+
+        function validateSdkSupport(sdkSupportObj, path) {
```

Comment:
I would consider removing `validateSdkSupport` and `checkRoot` methods as I don't think they are adding to the readability of the test.

Reply:
Since this is a recursive function, it is not easy to rewrite.

I have added some comments for clarification.